examples/SuperResolution/enet-pat.py

```diff
@@ -0,0 +1,310 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Author: Patrick Wieschollek <mail@patwie.com>
+
+import os
+import argparse
+import cv2
+import six
+import numpy as np
+import tensorflow as tf
+from tensorpack import *
+from tensorpack.tfutils.scope_utils import auto_reuse_variable_scope
+from GAN import MultiGPUGANTrainer, GANModelDesc
+from tensorpack.tfutils.summary import add_moving_summary
+from tensorpack.utils import logger
+from data_sampler import ImageDecode
+
+
+BATCH_SIZE = 6
+CHANNELS = 3
+SHAPE_LR = 32
+NF = 64
+VGG_MEAN = np.array([123.68, 116.779, 103.939])  # RGB
+
+
+def normalize(v):
+    assert isinstance(v, tf.Tensor)
+    v.get_shape().assert_has_rank(4)
+    dim = v.get_shape().as_list()
+    return v / (dim[1] * dim[2] * dim[3])
+
+
+def gram_matrix(v):
+    assert isinstance(v, tf.Tensor)
+    v.get_shape().assert_has_rank(4)
+    dim = v.get_shape().as_list()
+    v = normalize(v)
```

Comment:
The texture loss seems too small to be useful (there is already a very small loss weight for them). Does it really average over these dimensions here? The paper only mentions L2 norm.

Reply:
"For the perceptual loss LP and the texture loss LT , we
normalized feature activations to have a __mean__ of one [4]."